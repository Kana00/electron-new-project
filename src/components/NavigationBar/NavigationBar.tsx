import * as React from 'react';
import { Link } from 'react-router-dom';
import Flexbox from 'flexbox-react';

export default class NavigationBar extends React.Component<NavigationBarPropsType, any> {
  render() {
    const heightOfNavBar = 3;
    const styles = {
      navigationContainer: {
        /* Position Handler  */
        position: 'absolute',
        bottom: 0,
        /* Size Handler */
        width: '100vw',
        height: heightOfNavBar + 'rem',
      },
      styleLink: {
        flex: 1,
        height: heightOfNavBar + 'rem',
        textAlign: 'center',
        lineHeight: heightOfNavBar + 'rem',
        fontSize: heightOfNavBar / 3 + 'rem',
        backgroundColor: this.props.backgroundColor,
        color: this.props.textColor,
        textDecorationLine: 'none',
      }
    };
    return(
      <Flexbox flexDirection='row' justifyContent='space-between' alignItems='center' style={styles.navigationContainer}>
        <Link style={ styles.styleLink } to='/'>Main</Link>
        <Link style={ styles.styleLink } to='/featuresDemo'>Features</Link>
        <Link style={ styles.styleLink } to='/theme'>Theme</Link>
        <Link style={ styles.styleLink } to='/unknown'>NoMatch</Link>
      </Flexbox>
    );
  }
}
