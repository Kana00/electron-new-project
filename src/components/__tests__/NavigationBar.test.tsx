import * as React from 'react';
import NavigationBar from '../NavigationBar/NavigationBar';
import * as renderer from 'react-test-renderer';

jest.mock('react-router-dom');

describe('Test component NavigationBar', () => {
  it('render correctly', () => {
    const tree = renderer.create(<NavigationBar />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});
