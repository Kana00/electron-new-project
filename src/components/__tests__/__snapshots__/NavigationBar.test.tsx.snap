// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component NavigationBar render correctly 1`] = `
<div
  className="sc-bdVaJa cfPAtY"
  style={
    Object {
      "bottom": 0,
      "height": "3rem",
      "position": "absolute",
      "width": "100vw",
    }
  }
/>
`;
